src/java/org/apache/cassandra/cql3/statements/TransactionStatement.java

```diff
@@ -0,0 +1,440 @@
+                result.add(buffer);
+            }
+
+            return new ResultMessage.Rows(result.build());
+        }
+
+        // In the case of a write-only transaction, just return and empty result.
+        // TODO: This could be modified to return an indication of whether a condition (if present) succeeds.
+        return new ResultMessage.Void();
+    }
+
+    @Override
+    public ResultMessage executeLocally(QueryState state, QueryOptions options)
+    {
+        return execute(state, options, nanoTime());
+    }
+
+    @Override
+    public AuditLogContext getAuditLogContext()
+    {
+        return new AuditLogContext(AuditLogEntryType.TRANSACTION);
+    }
+
+    public static class Parsed extends QualifiedStatement
+    {
+        private final List<SelectStatement.RawStatement> assignments;
+        private final SelectStatement.RawStatement select;
+        private final List<RowDataReference.Raw> returning;
+        private final List<ModificationStatement.Parsed> updates;
+        private final List<ConditionStatement.Raw> conditions;
+        private final List<RowDataReference.Raw> dataReferences;
+
+        public Parsed(List<SelectStatement.RawStatement> assignments,
+                      SelectStatement.RawStatement select,
+                      List<RowDataReference.Raw> returning,
+                      List<ModificationStatement.Parsed> updates,
+                      List<ConditionStatement.Raw> conditions,
+                      List<RowDataReference.Raw> dataReferences)
+        {
+            super(null);
+            this.assignments = assignments;
+            this.select = select;
+            this.returning = returning;
+            this.updates = updates;
+            this.conditions = conditions != null ? conditions : Collections.emptyList();
+            this.dataReferences = dataReferences;
+        }
+
+        @Override
+        public void setKeyspace(ClientState state)
+        {
+            assignments.forEach(select -> select.setKeyspace(state));
+            updates.forEach(update -> update.setKeyspace(state));
+        }
+
+        @Override
+        public CQLStatement prepare(ClientState state)
+        {
+            checkFalse(updates.isEmpty() && returning == null && select == null, EMPTY_TRANSACTION_MESSAGE);
+
+            if (select != null || returning != null)
+                checkTrue(select != null ^ returning != null, "Cannot specify both a full SELECT and a SELECT w/ LET references.");
+
+            List<NamedSelect> preparedAssignments = new ArrayList<>(assignments.size());
+            Map<TxnDataName, RowDataReference.ReferenceSource> refSources = new HashMap<>();
+            Set<TxnDataName> selectNames = new HashSet<>();
+
+            for (SelectStatement.RawStatement select : assignments)
+            {
+                TxnDataName name = TxnDataName.user(select.parameters.refName);
+                checkNotNull(name, "Assignments must be named");
```

Comment:
name can't be null, we control that... you can check for `select.parameters.refName`